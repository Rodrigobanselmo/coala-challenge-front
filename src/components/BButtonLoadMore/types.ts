import { IBook } from "@/core/interfaces/IBook";
import { IPaginationResult } from "@/core/interfaces/IPaginationResult";
import { InfiniteData } from "@tanstack/react-query";
import { ElementType, ReactNode } from "react";

export interface ButtonLoadMoreListProps {
  fetchNextPage: () => void;
  canLoadMore: boolean;
  isFetchingNextPage: boolean;
}
