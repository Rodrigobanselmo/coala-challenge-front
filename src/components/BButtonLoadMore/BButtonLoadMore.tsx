import { FC } from "react";

import { Button } from "@mui/material";

import { ButtonLoadMoreListProps } from "./types";

export const BButtonLoadMore: FC<ButtonLoadMoreListProps> = ({
  canLoadMore,
  fetchNextPage,
  isFetchingNextPage,
}) => {
  return (
    <Button
      onClick={() => fetchNextPage()}
      variant="outlined"
      sx={{ width: 200 }}
    >
      {isFetchingNextPage
        ? "carregando..."
        : !canLoadMore
        ? "Mostrar mais"
        : "Não há mais livros"}
    </Button>
  );
};
