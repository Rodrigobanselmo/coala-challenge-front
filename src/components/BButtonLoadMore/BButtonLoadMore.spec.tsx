import React from "react";
import { render } from "@testing-library/react";
import { BButtonLoadMore } from "./BButtonLoadMore";

const mockProps = {
  canLoadMore: true,
  fetchNextPage: () => {},
  isFetchingNextPage: false,
};

describe("BButtonLoadMore", () => {
  it("matches the snapshot when can load more", () => {
    const { container } = render(<BButtonLoadMore {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot when fetching next page", () => {
    const { container } = render(
      <BButtonLoadMore {...mockProps} isFetchingNextPage={true} />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot when cannot load more", () => {
    const { container } = render(
      <BButtonLoadMore {...mockProps} canLoadMore={false} />
    );
    expect(container).toMatchSnapshot();
  });
});
