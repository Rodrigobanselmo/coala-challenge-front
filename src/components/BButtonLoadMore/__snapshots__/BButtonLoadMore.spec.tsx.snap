// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BButtonLoadMore matches the snapshot when can load more 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Não há mais livros
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`BButtonLoadMore matches the snapshot when cannot load more 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Mostrar mais
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`BButtonLoadMore matches the snapshot when fetching next page 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    carregando...
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
